opslevel/datasource_opslevel_lifecycles.go

```diff
@@ -1,63 +1,94 @@
-// )
-
-// func datasourceLifecycles() *schema.Resource {
-// 	return &schema.Resource{
-// 		Read: wrap(datasourceLifecyclesRead),
-// 		Schema: map[string]*schema.Schema{
-// 			"aliases": {
-// 				Type:     schema.TypeList,
-// 				Computed: true,
-// 				Elem:     &schema.Schema{Type: schema.TypeString},
-// 			},
-// 			"ids": {
-// 				Type:     schema.TypeList,
-// 				Computed: true,
-// 				Elem:     &schema.Schema{Type: schema.TypeString},
-// 			},
-// 			"indexes": {
-// 				Type:     schema.TypeList,
-// 				Computed: true,
-// 				Elem:     &schema.Schema{Type: schema.TypeInt},
-// 			},
-// 			"names": {
-// 				Type:     schema.TypeList,
-// 				Computed: true,
-// 				Elem:     &schema.Schema{Type: schema.TypeString},
-// 			},
-// 		},
-// 	}
-// }
-
-// func datasourceLifecyclesRead(d *schema.ResourceData, client *opslevel.Client) error {
-// 	result, err := client.ListLifecycles()
-// 	if err != nil {
-// 		return err
-// 	}
-
-// 	count := len(result)
-// 	aliases := make([]string, count)
-// 	ids := make([]string, count)
-// 	indexes := make([]int, count)
-// 	names := make([]string, count)
-// 	for _, item := range result {
-// 		i := item.Index - 1
-// 		aliases[i] = item.Alias
-// 		ids[i] = string(item.Id)
-// 		indexes[i] = item.Index
-// 		names[i] = item.Name
-// 	}
-
-// 	d.SetId(timeID())
-// 	d.Set("aliases", aliases)
-// 	d.Set("ids", ids)
-// 	d.Set("indexes", indexes)
-// 	d.Set("names", names)
-
-// 	return nil
-// }
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+// Ensure LifecycleDataSource implements DataSourceWithConfigure interface
+var _ datasource.DataSourceWithConfigure = &LifecycleDataSource{}
+
+func NewLifecycleDataSourcesAll() datasource.DataSource {
+	return &LifecycleDataSourcesAll{}
+}
+
+// LifecycleDataSource manages a Lifecycle data source.
+type LifecycleDataSourcesAll struct {
+	CommonDataSourceClient
+}
+
+// CategoryDataSourcesAll manages a Category data source.
+type LifecycleDataSourcesAllModel struct {
+	Lifecycles []lifecycleDataSourceModel `tfsdk:"lifecycles"`
+}
+
+// lifecycleDataSourceModel describes the data source data model.
+type lifecycleDataSourceModel struct {
+	Alias types.String `tfsdk:"alias"`
+	Id    types.String `tfsdk:"id"`
+	Index types.Int64  `tfsdk:"index"`
+	Name  types.String `tfsdk:"name"`
+}
+
+func NewLifecycleDataSourcesAllModel(lifecycles []opslevel.Lifecycle) LifecycleDataSourcesAllModel {
+	lifecyclesModel := []lifecycleDataSourceModel{}
+	for _, lifecycle := range lifecycles {
+		lifecycleModel := lifecycleDataSourceModel{
+			Alias: ComputedStringValue(lifecycle.Alias),
+			Id:    ComputedStringValue(string(lifecycle.Id)),
+			Index: types.Int64Value(int64(lifecycle.Index)),
```

Comment:
We should add a `ComputedInt64Value` function

Reply:
Sure! I'll add something in a separate PR soon